src/main/java/nl/tudelft/cse1110/andy/result/ResultBuilder.java

```diff
@@ -292,8 +311,15 @@ private int calculateFinalGrade(GradeValues grades, GradeWeight weights) {
     public boolean hasFailed() {
         boolean compilationFailed = compilation!=null && !compilation.successful();
         boolean unitTestsFailed = testResults != null && testResults.didNotGoWell();
-        boolean genericFailureHappened = genericFailureMessage != null;
+        boolean hasGenericFailure = hasGenericFailure();
+
+        return compilationFailed || unitTestsFailed || hasGenericFailure;
+    }
+
+    private boolean hasGenericFailure() {
```

Comment:
Also thinking of a way to move this method to the new GenericFailure class

Reply:
I made `GenericFailure.build()` return null if the arguments provided to it mean that there is no generic failure (i.e. they are all null). This makes the `hasGenericFailure` method obsolete, so I removed it entirely.